crates/rpc/rpc/src/eth/revm_utils.rs

```diff
@@ -374,23 +374,19 @@ pub(crate) struct CallFees {
 impl CallFees {
     /// Ensures the fields of a [CallRequest] are not conflicting.
     ///
-    /// If no `gasPrice` or `maxFeePerGas` is set, then the `gas_price` in the response will
-    /// fallback to the given `basefee`.
+    /// If no `gasPrice` or `maxFeePerGas` is set, then the `gas_price` in the returned `gas_price`
+    /// will be `0`. See: <https://github.com/ethereum/go-ethereum/blob/2754b197c935ee63101cbbca2752338246384fec/internal/ethapi/transaction_args.go#L242-L255>
     fn ensure_fees(
         call_gas_price: Option<U256>,
         call_max_fee: Option<U256>,
         call_priority_fee: Option<U256>,
         base_fee: U256,
     ) -> EthResult<CallFees> {
         match (call_gas_price, call_max_fee, call_priority_fee) {
-            (None, None, None) => {
-                // when none are specified, they are all set to zero
-                Ok(CallFees { gas_price: U256::ZERO, max_priority_fee_per_gas: None })
-            }
             (gas_price, None, None) => {
-                // request for a legacy transaction
-                // set everything to zero
-                let gas_price = gas_price.unwrap_or(base_fee);
+                // either legacy or no fee fields
+                // when none are specified, they are all set to zero
```

Comment:
this comment still confuses me a bit, maybe something like this?
```suggestion
                // either legacy transaction or no fee fields are specified
                // when no fields are specified, set gas price to zero
```